hashicorp/terraform-ls
# Replace internal watcher with LSP's watching mechanism

Fixes #867 

--- 

## How to test this

I added two E2E tests confirming both provider schema & module manifest refresh is working. The previous watcher didn't have any tests (since mocking `fsnotify` events was not trivial).

One can also check the module & provider views in the VS Code extension as they update automatically after `terraform init`:

https://user-images.githubusercontent.com/287584/174062323-a4a6dce5-9c0d-44c2-8ad4-e5a4ed676a6d.mov
